app/controllers/user.js

```diff
@@ -0,0 +1,39 @@
+const { validationResult } = require('express-validator/check');
+const bcrypt = require('bcryptjs');
+const salt = bcrypt.genSaltSync(10);
+const { databaseError } = require('../errors');
+const { bad_request_error } = require('../errors');
+const { conflict_error } = require('../errors');
+const userModel = require('../models').user;
+
+exports.register = (req, res, next) => {
+  const errors = validationResult(req);
+  if (errors.isEmpty() === false) {
+    console.log(errors);
+    next(bad_request_error(errors.array()));
+  } else {
+    userModel
+      .findOne({ where: { email: req.body.email } })
+      .then(user => {
+        if (user) {
+          next(conflict_error('user already exists!'));
```

Comment:
409 status code seems don't apply in this case `https://httpstatuses.com/409`

Reply:
As I saw in different forums the 409 status code is the most accurate to use in this case. I could use 400 status code too.
https://stackoverflow.com/questions/1434315/http-status-code-for-database-is-down

Reply:
Nice